Framework/HistogramData/src/Rebin.cpp

```diff
@@ -0,0 +1,147 @@
+#include "MantidHistogramData/Rebin.h"
+#include "MantidHistogramData/BinEdges.h"
+#include "MantidHistogramData/Histogram.h"
+#include <algorithm>
+#include <numeric>
+
+using Mantid::HistogramData::Histogram;
+using Mantid::HistogramData::BinEdges;
+using Mantid::HistogramData::Counts;
+using Mantid::HistogramData::CountStandardDeviations;
+using Mantid::HistogramData::CountVariances;
+using Mantid::HistogramData::Frequencies;
+using Mantid::HistogramData::FrequencyStandardDeviations;
+using Mantid::HistogramData::FrequencyVariances;
+
+namespace {
+Histogram rebinCounts(const Histogram &input, const BinEdges &binEdges) {
+  auto &xold = input.x();
+  auto &yold = input.y();
+  auto &eold = input.e();
+
+  auto &xnew = binEdges.rawData();
+  Counts newCounts(xnew.size() - 1);
+  CountVariances newCountVariances(xnew.size() - 1);
+  auto &ynew = newCounts.mutableData();
+  auto &enew = newCountVariances.mutableData();
+
+  auto size_yold = yold.size();
+  auto size_ynew = ynew.size();
+  size_t iold = 0;
+  size_t inew = 0;
+
+  while ((inew < size_ynew) && (iold < size_yold)) {
+    auto xo_low = xold[iold];
+    auto xo_high = xold[iold + 1];
+    auto xn_low = xnew[inew];
+    auto xn_high = xnew[inew + 1];
+    auto owidth = xo_high - xo_low;
+    auto nwidth = xn_high - xn_low;
+
+    if (owidth <= 0.0 || nwidth <= 0.0)
+      throw std::runtime_error("Negative or zero bin widths not allowed.");
+
+    if (xn_high <= xo_low)
+      inew++; /* old and new bins do not overlap */
+    else if (xo_high <= xn_low)
+      iold++; /* old and new bins do not overlap */
+    else {
+      // delta is the overlap of the bins on the x axis
+      auto delta = xo_high < xn_high ? xo_high : xn_high;
+      delta -= xo_low > xn_low ? xo_low : xn_low;
```

Comment:
The [`VectorHelper`](https://github.com/mantidproject/mantid/blob/master/Framework/Kernel/src/VectorHelper.cpp#L160) code has a check for `delta < 0` to indicate no overlap. This seems to be missing here.

Is that the case or have I misunderstood something about checking elsewhere?